### DISCUSSION TITLE
how to setup hooks in mocha framework?

### DISCUSSION THREAD

**Discussion by garyzhang21** [webdriverio/webdriverio#7629] [Q&A]
HI, i want to use beforeEach,afterEach,before,after as the hooks in Mocha test case level. but now it seems like we need to setup at config file.
is it possible to provide it in case level??

**Comment by garyzhang21** [comment]
![image](https://user-images.githubusercontent.com/32188719/139801274-6c290513-9b2a-4652-b76d-018451a21510.png)

**Reply by garyzhang21** [reply]
now we can set up hook in wdio.conf.js like the picture 
![image](https://user-images.githubusercontent.com/32188719/140710478-25de3deb-f24b-4644-a914-3edd5372f791.png)
but i want to use the hooks in mocha test case(before,beforeeach,after,afterEach),what should i do?

**Reply by christian-bromann** [reply]
just put them in a Mocha testcase, e.g.

```js
beforeEach(() => {
  // ...
})

describe('foobar', () => {

    before(() => {
      // ...
    })

    // ...
})
```

**Reply by garyzhang21** [reply]
i tried it . but beforeEach(() => {
  // ...
}) does not work. it will skip. so do i need to add some setting to wdio.conf.js file?

**Reply by christian-bromann** [reply]
`beforeEach` is a [Mocha hook](https://mochajs.org/#hooks) that should work.

**Reply by garyzhang21** [reply]
yes, so i wonder if we have some config setting in wdio..

**Reply by christian-bromann** [reply]
What do you mean?
